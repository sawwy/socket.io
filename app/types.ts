export type UserType = {
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
  username: string;
  avatar: string;
  description: string;
};

export interface UserResponseType
  extends Omit<UserType, "lastSeen" | "joined"> {
  lastSeen: string;
  joined: string;
}

export type SelectedUserDataType = {
  user: UserType;
  rect: DOMRect;
};
