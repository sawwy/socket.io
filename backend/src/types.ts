import { MessageTypeEnum } from "./enums.ts";

export interface IUser {
  id: string;
  username: string;
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
}

export interface IUserMessage {
  messageType: MessageTypeEnum.UserMessage;
  message: string;
  user: IUser;
  timestamp: string;
}

export interface IUsers {
  [userId: string]: IUser;
}
