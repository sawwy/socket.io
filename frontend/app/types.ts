export interface IUser {
  id: string;
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
  username: string;
  avatar?: string;
  description?: string;
}

export interface UserResponseType extends Omit<IUser, "lastSeen" | "joined"> {
  lastSeen: string;
  joined: string;
}

export type SelectedUserDataType = {
  user: IUser;
  rect: DOMRect;
};

export interface IMessage {
  message: string;
  username: string;
  timestamp: string;
}
