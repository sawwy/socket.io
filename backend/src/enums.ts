export enum MessageTypeEnum {
  UserMessage = "userMessage",
  SystemMessage = "systemMessage",
}
