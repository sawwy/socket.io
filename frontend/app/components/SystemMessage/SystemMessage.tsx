import { ISystemMessage } from "~/types";
import styles from "./styles.module.css";

type SystemMessagePropsType = {
  message: ISystemMessage;
};

export const SystemMessage = ({ message }: SystemMessagePropsType) => {
  return (
    <div data-testid="chat-systemmessage" className={styles.message}>
      <p>{message.message}</p>
    </div>
  );
};
