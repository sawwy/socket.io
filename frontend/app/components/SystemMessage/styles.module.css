.message {
  color: rgb(210, 213, 217);
}
