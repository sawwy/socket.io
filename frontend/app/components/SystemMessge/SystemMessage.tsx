import { ISystemMessage } from "~/types";
import styles from "./styles.module.css";

type SystemMessagePropsType = {
  message: ISystemMessage;
};

export const SystemMessage = ({ message }: SystemMessagePropsType) => {
  return (
    <div className={styles.message}>
      <p>{message.message}</p>
    </div>
  );
};
