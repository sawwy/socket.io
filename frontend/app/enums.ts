export enum UsernameErrors {
  TooShort = "Username must be longer than 3 characters",
}
