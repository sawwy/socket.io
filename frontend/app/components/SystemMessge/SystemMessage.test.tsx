import renderer from "react-test-renderer";
import { SystemMessage } from "./SystemMessage";
import { ISystemMessage, IUser } from "~/types";
import { MessageTypeEnum } from "~/enums";

const message: ISystemMessage = {
  type: MessageTypeEnum.SystemMessage,
  message: "Qwerty has joined the lobby",
};

describe("SystemMessage Snapshot", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer
      .create(<SystemMessage message={message} />)
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
