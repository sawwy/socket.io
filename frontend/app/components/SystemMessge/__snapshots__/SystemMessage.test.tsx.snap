// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemMessage Snapshot Matches DOM Snapshot 1`] = `
<div
  className="message"
  data-testid="chat-systemmessage"
>
  <p>
    Qwerty has joined the lobby
  </p>
</div>
`;
